server/game/cards/20-HMW/EliasLance.js

```diff
@@ -0,0 +1,41 @@
+const DrawCard = require('../../drawcard.js');
+
+class EliasLance extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction({ trait: 'Sand Snake' });
+
+        this.whileAttached({
+            match: card => card.name === 'Elia Sand',
+            effect: ability.effects.addKeyword('Stealth')
+        });
+
+        this.reaction({
+            when: {
+                afterChallenge: event => event.challenge.loser === this.controller && this.parent.isAttacking()
+            },
+            handler: context => {
+                this.game.promptForSelect(context.event.challenge.winner, {
+                    activePromptTitle: 'Select a defending character',
+                    source: this,
+                    cardCondition: card => card.location === 'play area' &&
```

Comment:
Since the card uses the word "choose", this should be defined as a target, and use the `choosingPlayer` property to make the winner choose (this might require modifying what `choosingPlayer` supports)